modules/channel-vonage/src/renderers/template-media.ts

```diff
@@ -0,0 +1,52 @@
+import { ChannelRenderer } from 'common/channel'
+import { CHANNEL_NAME } from '../backend/client'
+import { TemplateComponents, Parameters, Buttons } from '../backend/templates'
+import { ChannelContentCustomTemplate, TemplateLanguage, VonageContext } from '../backend/typings'
+
+export class VonageMediaTemplateRenderer implements ChannelRenderer<VonageContext> {
+  get channel(): string {
+    return CHANNEL_NAME
+  }
+
+  get priority(): number {
+    return 0
+  }
+
+  get id(): string {
+    return VonageMediaTemplateRenderer.name
+  }
+
+  handles(context: VonageContext): boolean {
+    return context.payload.type === 'vonage-media-template'
+  }
+
+  async render(context: VonageContext) {
+    const payload = context.payload
+
+    const headerParameters = (payload.header?.parameters || []) as Parameters
+    const bodyParameters = (payload.body?.parameters || []) as Parameters
+    const buttonParameters = (payload.button?.parameters || []) as Buttons
+
+    const components = new TemplateComponents()
+      .withHeader(...headerParameters)
+      .withBody(...bodyParameters)
+      .withButtons(...buttonParameters)
+      .build()
```

Comment:
In my opinion we're implementing too much custom logic here. Why not let the content renderers provide the whole `template` property? Channel renderers are meant to convert standarized payloads to channel specific payloads. Templates are already channel specific to vonage so it's fine to construct the whole payload in the content renderer

Reply:
I totally agree with you! I think we should simply remove the component builder. Since we are using 'static' content-types, there is no need to have this kind of building pattern here. I'll make the changes!